src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.Android.cs

```diff
@@ -371,7 +371,11 @@ private void UpdateInputScope(InputScope inputScope)
 
 		partial void OnTextWrappingChangedPartial(DependencyPropertyChangedEventArgs e)
 		{
-			//TODO : see bug #8178
+			if (_textBoxView != null && e.NewValue is TextWrapping textWrapping)
+			{
+				var needsWrapping = textWrapping is TextWrapping.Wrap || textWrapping is TextWrapping.WrapWholeWords;
+				_textBoxView.SetSingleLine(!needsWrapping);
```

Comment:
Same thing but a bit smaller and more readable (IMO)

```suggestion
				_textBoxView.SetSingleLine(textWrapping == TextWrapping.NoWrap);
```

I also think we need to do the same in:

https://github.com/unoplatform/uno/blob/5e0c3818f1badff1f148325e7c3ff7b53647d4a1/src/Uno.UI/UI/Xaml/Controls/TextBox/TextBoxView.Android.cs#L45